src/mscorlib/shared/System/Diagnostics/Debug.Unix.cs

```diff
@@ -21,7 +21,19 @@ private static void ShowDialog(string stackTrace, string message, string detailM
                 // In Core, we do not show a dialog.
                 // Fail in order to avoid anyone catching an exception and masking
                 // an assert failure.
-                var ex = new DebugAssertException(message, detailMessage, stackTrace);
+                DebugAssertException ex;
+                if (message == String.Empty) 
```

Comment:
Also, what if the user passes in null for these? Should there be string.IsNullOrEmpty? IsNullOrWhiteSpace?